### DISCUSSION TITLE
Where is the part related to assist mode in the firmware?

### DISCUSSION THREAD

**Discussion by dixxLee** [bitcraze/discussions#499] [Q&A] [answered]
Hello.

In the cfclient, I want to know which part of the firmware code is the code that recognizes when a button is pressed on the remote controller for assist mode.
I wonder where the part that receives the operation of assist mode is.

**Comment by krichardsson** [comment, marked as answer]
This is "client only" functionality, the only difference is the kind of setpoints that are sent to the Crazyflie. 

The central part of the code in the client is where the [correct type of setpoint is selected](https://github.com/bitcraze/crazyflie-clients-python/blob/09eed5adbaa175ea4598db2846f31dabaa8aa889/src/cfclient/ui/main.py#L196-L209)

In the firmware you can find the handling of different types of setpoints in the [crtp_commander_generic.c](https://github.com/bitcraze/crazyflie-firmware/blob/master/src/modules/src/crtp_commander_generic.c) file

**Reply by dixxLee** [reply]
Your answer was very helpful. thanks.
